### DISCUSSION TITLE
How to confirm that the asyn projection is completed before querying

### DISCUSSION THREAD

**Discussion by yk442755675** [JasperFx/marten#2490] [Q&A]
Hello all,
I'm doing a async projection when i save a event. but at same time, the system will do query to get the projection. 
the question is when i doing query to get data, sometime it has the projection, sometime doesn't.
i saw some way to handle this question, but i want know waht is the best way to handle this situation.

**Comment by legendxian** [comment]
I had this same question!

**Comment by gfoidl** [comment]
Using an inline or live projection would be the correct approach to this.

As by design and definition the async projection is async, so it's pure luck that the projection is done when you query immediately (race condition). 
Depending on your use-case something like Postgres notification could be used to get that information. But as said, IMO this is the wron approach here.
